arbos/constraints/multi_dimensional.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2025, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+// The constraints package tracks the multi-dimensional gas usage to apply constraint-based pricing.
+package constraints
+
+import (
+	"github.com/ethereum/go-ethereum/arbitrum/multigas"
+
+	"github.com/offchainlabs/nitro/arbos/storage"
+	"github.com/offchainlabs/nitro/util/arbmath"
+)
+
+// Fixed flat layout for a Multi-Constraint:
+// [0] target (uint64)
+// [1] adjustmentWindow (uint64)
+// [2] backlog (uint64)
+// [3..3+NumResourceKind-1] weighted resources (uint64 each)
+
+const (
+	targetOffset uint64 = iota
+	adjustmentWindowOffset
+	backlogOffset
+	weightedResourcesBaseOffset
+)
+
+// MultiGasConstraint defines a pricing constraint that combines several
```

Comment:
I'm wondering if there is a strong reason to not use a RLP encoding for the constraints.